src/pyrokinetics/gk_code/gs2.py

```diff
@@ -312,30 +332,63 @@ def _read_single_grid(self):
             "theta0": theta0,
         }
 
-    def _read_range_grid(self):
+    def _read_range_grid(self, local_geometry):
         range_options = self.data["kt_grids_range_parameters"]
-        nky = range_options.get("naky", 1)
 
-        ky_min = range_options.get("aky_min", 0.0)
-        ky_max = range_options.get("aky_max", 0.0)
+        if "nn0" in range_options.keys():
+            drho_dpsi = (
```

Comment:
Duplication wrt single -- can you pull this out into a separate method?

Reply:
Moved to `read_grids` and pass `drho_dpsi` to the different methods